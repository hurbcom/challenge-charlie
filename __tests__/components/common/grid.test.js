import React from 'react';
import renderer from 'react-test-renderer';

//Components
import {
    Grid1,
    Grid10,
    Grid11,Grid12,
    Grid2,
    Grid3,
    Grid4,
    Grid5,
    Grid6,
    Grid7,
    Grid8,
    Grid9,GridNumber
} from "../../../extra/components/common/grid";

test('Grid1 Component', () => {
    const component = renderer.create(
        <Grid1>
            <div id={'test'}/>
        </Grid1>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid2 Component', () => {
    const component = renderer.create(
        <Grid2>
            <div id={'test'}/>
        </Grid2>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid3 Component', () => {
    const component = renderer.create(
        <Grid3>
            <div id={'test'}/>
        </Grid3>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid4 Component', () => {
    const component = renderer.create(
        <Grid4>
            <div id={'test'}/>
        </Grid4>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid5 Component', () => {
    const component = renderer.create(
        <Grid5>
            <div id={'test'}/>
        </Grid5>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid6 Component', () => {
    const component = renderer.create(
        <Grid6>
            <div id={'test'}/>
        </Grid6>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid7 Component', () => {
    const component = renderer.create(
        <Grid7>
            <div id={'test'}/>
        </Grid7>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid8 Component', () => {
    const component = renderer.create(
        <Grid8>
            <div id={'test'}/>
        </Grid8>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid9 Component', () => {
    const component = renderer.create(
        <Grid9>
            <div id={'test'}/>
        </Grid9>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid10 Component', () => {
    const component = renderer.create(
        <Grid10>
            <div id={'test'}/>
        </Grid10>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid11 Component', () => {
    const component = renderer.create(
        <Grid11>
            <div id={'test'}/>
        </Grid11>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('Grid12 Component', () => {
    const component = renderer.create(
        <Grid12>
            <div id={'test'}/>
        </Grid12>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
});

test('GridNumber Component', () => {
    const component = renderer.create(
        <GridNumber number={1}>
            <div id={'test'}/>
        </GridNumber>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');
    //Value props children
    expect(tree.props.number).toEqual(1);
});