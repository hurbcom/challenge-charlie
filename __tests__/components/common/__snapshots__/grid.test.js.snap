// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid1 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid2 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2 MuiGrid-grid-xl-2"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid3 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid4 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid5 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-5 MuiGrid-grid-sm-5 MuiGrid-grid-md-5 MuiGrid-grid-lg-5 MuiGrid-grid-xl-5"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid6 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid7 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 MuiGrid-grid-lg-7 MuiGrid-grid-xl-7"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid8 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-md-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid9 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-sm-9 MuiGrid-grid-md-9 MuiGrid-grid-lg-9 MuiGrid-grid-xl-9"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid10 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-10 MuiGrid-grid-md-10 MuiGrid-grid-lg-10 MuiGrid-grid-xl-10"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid11 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-11 MuiGrid-grid-md-11 MuiGrid-grid-lg-11 MuiGrid-grid-xl-11"
>
  <div
    id="test"
  />
</div>
`;

exports[`Grid12 Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
>
  <div
    id="test"
  />
</div>
`;

exports[`GridNumber Component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
  number={1}
>
  <div
    id="test"
  />
</div>
`;
